nodebuilder/share/module.go

```diff
@@ -276,3 +291,70 @@ func lightAvailabilityComponents(cfg *Config) fx.Option {
 		}),
 	)
 }
+
+func archivalComponents(cfg *Config) fx.Option {
+	return fx.Options(
+		// archival service discovery
+		fx.Provide(func(
+			lc fx.Lifecycle,
+			pruneCfg *pruner.Config,
+			d *disc.Discovery,
+			h host.Host,
+			r routing.ContentRouting,
+			opt disc.Option,
+		) ([]*disc.Discovery, error) {
+			// if pruner is enabled, no archival service is necessary
```

Comment:
Don't pruned nodes at least need archivalDisc?

Reply:
here it is: https://github.com/celestiaorg/celestia-node/pull/3188/commits/9604b96b7c9aa6fcf7fbfca31db11f4906487584